lib/store_model/types/array_type.rb

```diff
@@ -69,15 +70,19 @@ def changed_in_place?(raw_old_value, new_value)
       # rubocop:disable Style/RescueModifier
       def decode_and_initialize(array_value)
         decoded = ActiveSupport::JSON.decode(array_value) rescue []
-        decoded.map { |attributes| @model_klass.new(attributes) }
+        decoded.map { |attributes| cast_model_type_value(attributes) }
       end
       # rubocop:enable Style/RescueModifier
 
       def ensure_model_class(array)
         array.map do |object|
-          object.is_a?(@model_klass) ? object : @model_klass.new(object)
+          object.is_a?(@model_klass) ? object : cast_model_type_value(object)
         end
       end
+
+      def cast_model_type_value(value)
+        @model_klass_type.cast_value(value)
```

Comment:
Is this enough for `handle_unknown_attribute` with `rescue` block to be called?  to be called?

Reply:
Yep, having the JsonType instance generated for the current `model_class` we can call `cast_value` on it, so it will handle the remaining work for us (rescue the exception and collect unknown attributes)